stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/facotry/AttachmentsPickerMediaCaptureTabFactory.kt

```diff
@@ -0,0 +1,131 @@
+    override val attachmentsPickerMode: AttachmentsPickerMode
+        get() = MediaCapture
+
+    /**
+     * Emits a camera icon for this tab.
+     *
+     * @param isEnabled If the tab is enabled.
+     * @param isSelected If the tab is selected.
+     */
+    @Composable
+    override fun pickerTabIcon(isEnabled: Boolean, isSelected: Boolean) {
+        Icon(
+            painter = painterResource(id = R.drawable.stream_compose_ic_media_picker),
+            contentDescription = stringResource(id = R.string.stream_compose_capture_option),
+            tint = when {
+                isSelected -> ChatTheme.colors.primaryAccent
+                isEnabled -> ChatTheme.colors.textLowEmphasis
+                else -> ChatTheme.colors.disabled
+            },
+        )
+    }
+
+    /**
+     * Emits a content for this tab.
+     *
+     * @param attachments The list of attachments to display.
+     * @param onAttachmentsChanged Handler to set the loaded list of attachments to display.
+     * @param onAttachmentItemSelected Handler when the item selection state changes.
+     * @param onAttachmentsSubmitted Handler to submit the selected attachments to the message composer.
+     */
+    @OptIn(ExperimentalPermissionsApi::class)
+    @Composable
+    override fun pickerTabContent(
+        attachments: List<AttachmentPickerItemState>,
+        onAttachmentsChanged: (List<AttachmentPickerItemState>) -> Unit,
+        onAttachmentItemSelected: (AttachmentPickerItemState) -> Unit,
+        onAttachmentsSubmitted: (List<AttachmentMetaData>) -> Unit,
+    ) {
+        val context = LocalContext.current
+
+        val requiresCameraPermission = isCameraPermissionDeclared(context)
+
+        val cameraPermissionState =
+            if (requiresCameraPermission) rememberPermissionState(permission = Manifest.permission.CAMERA) else null
+
+        val mediaCaptureResultLauncher =
+            rememberLauncherForActivityResult(contract = CaptureMediaContract()) { file: File? ->
+                val attachments = if (file == null) {
+                    emptyList()
+                } else {
+                    listOf(AttachmentMetaData(context, file))
+                }
+
+                onAttachmentsSubmitted(attachments)
+            }
+
+        if (cameraPermissionState == null || cameraPermissionState.status == PermissionStatus.Granted) {
+            Box(modifier = Modifier.fillMaxSize()) {
+                LaunchedEffect(Unit) {
+                    mediaCaptureResultLauncher.launch(Unit)
+                }
+            }
+        } else if (cameraPermissionState.status is PermissionStatus.Denied) {
+            MissingPermissionContent(cameraPermissionState)
+        }
+    }
+}
+
+/**
+ * Returns if we need to check for the camera permission or not.
+ *
+ * @param context The context of the app.
+ * @return If the camera permission is declared in the manifest or not.
+ */
+private fun isCameraPermissionDeclared(context: Context): Boolean {
```

Comment:
Could we move this inside the factory? No reason to have it as top-level, if it's private and not used outside